src/webgpu/api/operation/command_buffer/render/index_format.spec.ts

```diff
@@ -0,0 +1,299 @@
+        gl_Position = pos;
+    }`;
+
+    const fragmentShaderGLSL = `#version 450
+    layout(location = 0) out vec4 fragColor;
+    void main() {
+        fragColor = vec4(0.0, 1.0, 0.0, 1.0);
+    }`;
+
+    return this.device.createRenderPipeline({
+      layout: this.device.createPipelineLayout({ bindGroupLayouts: [] }),
+
+      vertexStage: {
+        module: this.device.createShaderModule({
+          code: this.glslang.compileGLSL(vertexShaderGLSL, 'vertex'),
+        }),
+        entryPoint: 'main',
+      },
+      fragmentStage: {
+        module: this.device.createShaderModule({
+          code: this.glslang.compileGLSL(fragmentShaderGLSL, 'fragment'),
+        }),
+        entryPoint: 'main',
+      },
+
+      primitiveTopology: 'triangle-strip',
+
+      colorStates: [
+        {
+          format: this.swapChainFormat,
+        },
+      ],
+
+      vertexState: {
+        indexFormat: format,
+        vertexBuffers: [
+          {
+            arrayStride: 4 * 4,
+            stepMode: 'vertex',
+            attributes: [
+              {
+                format: 'float4',
+                offset: 0,
+                shaderLocation: 0,
+              },
+            ],
+          },
+        ],
+      },
+    });
+  }
+
+  MakeBufferMapped(arrayBuffer: ArrayBuffer, usage: number): GPUBuffer {
+    const [buffer, bufferMapping] = this.device.createBufferMapped({
+      size: arrayBuffer.byteLength,
+      usage,
+    });
+
+    if (arrayBuffer instanceof Float32Array) {
+      new Float32Array(bufferMapping).set(arrayBuffer);
+    } else if (arrayBuffer instanceof Uint32Array) {
+      new Uint32Array(bufferMapping).set(arrayBuffer);
+    } else if (arrayBuffer instanceof Uint16Array) {
+      new Uint16Array(bufferMapping).set(arrayBuffer);
+    }
+
+    buffer.unmap();
+    return buffer;
+  }
+
+  run(
+    vertexBuffer: GPUBuffer,
+    indexBuffer: GPUBuffer,
+    indexCount: number,
+    expected: Uint8Array
+  ): void {
+    const encoder = this.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: this.colorAttachment.createView(),
+          loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+          storeOp: 'store',
+        },
+      ],
+    });
+    pass.setPipeline(this.pipeline);
+    pass.setVertexBuffer(0, vertexBuffer);
+    pass.setIndexBuffer(indexBuffer);
+    pass.drawIndexed(indexCount, 1, 0, 0, 0);
+    pass.endPass();
+    encoder.copyTextureToBuffer(
+      { texture: this.colorAttachment, mipLevel: 0, origin: { x: 51, y: 51, z: 0 } },
+      { buffer: this.result, bytesPerRow: 256 },
+      { width: 1, height: 1, depth: 1 }
+    );
+    this.device.defaultQueue.submit([encoder.finish()]);
+
+    this.expectContents(this.result, expected);
+  }
+}
+
+export const g = new TestGroup(IndexFormatTest);
+
+g.test('uint32').fn(async t => {
+  await t.initResources('uint32');
+
+  const vertexArray = new Float32Array([
```

Comment:
Consider putting /* prettier-ignore */  on this so it doesn't autoformat so poorly